csrc/scheduler/matmul.cpp

```diff
@@ -880,25 +881,17 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   bcw_smem->definition()->as<LoadStoreOp>()->setCacheOp(cache_op_b);
   NVF_ERROR(acw_smem->uses().size() == 1);
   NVF_ERROR(bcw_smem->uses().size() == 1);
-  if (auto ldst = dynamic_cast<LoadStoreOp*>(acw_smem->uses().at(0))) {
-    acr = ldst->out()->as<TensorView>();
-    if (ldst->hasInnerTranspose()) {
-      ldst->setOpType(LoadStoreOpType::LdMatrixTranspose);
-    } else {
-      ldst->setOpType(LoadStoreOpType::LdMatrix);
-    }
-  } else {
-    acr = acw_smem->cacheAfter(LoadStoreOpType::LdMatrix);
-  }
-  if (auto ldst = dynamic_cast<LoadStoreOp*>(bcw_smem->uses().at(0))) {
-    bcr = ldst->out()->as<TensorView>();
-    if (ldst->hasInnerTranspose()) {
-      ldst->setOpType(LoadStoreOpType::LdMatrixTranspose);
+
+  for (auto [tv_smem, tv_r] :
+       {std::make_pair(acw_smem, &acr), std::make_pair(bcw_smem, &bcr)}) {
+    if (auto ldst = dynamic_cast<LoadStoreOp*>(tv_smem->uses().at(0))) {
+      *tv_r = ldst->out()->as<TensorView>();
     } else {
-      ldst->setOpType(LoadStoreOpType::LdMatrix);
+      *tv_r = tv_smem->cacheAfter(
+          LoadStoreOpType::Set,
+          CacheOp::Unspecified,
+          false /* propagate allocation*/);
```

Comment:
Passing on a tip I got from @wujingyue: if you use `/*propagate_allocation_domain=*/false` instead then `clang-tidy` will check that the variable name matches the comment.

Reply:
Please add comments about the allocation domains of `tv_smem` and `tv_r`, and why you need to not propagate allocation domain at this location.